db/migrate/20150716163704_remove_filesystem_repositories.rb

```diff
@@ -30,12 +30,37 @@
 # Removes all remaining Repository::Filesystem entries.
 #
 class RemoveFilesystemRepositories < ActiveRecord::Migration
+  include Migration::Utils
+
   def up
+    # Delete any changesets belonging to filesystem repositories
+    if postgres?
+      ActiveRecord::Base.connection.execute <<-SQL
+        DELETE FROM changesets
+        WHERE repository_id IN (
+          SELECT id
+          FROM repositories
+          WHERE type = #{filesystem_type}
+        )
```

Comment:
Since the current form for postgres is the more portable one (I believe this is completely ANSI SQL) I would propose to always use it.

Not that we currently intend to support anything else but PSQL and MySQL, I usually prefer generic solutions. If only for the reduced duplication and its problems.